default/python/AI/pathfinding.py

```diff
@@ -0,0 +1,55 @@
+from heapq import heappush, heappop
+from collections import namedtuple
+
+import freeOrionAIInterface as fo
+
+
+def find_path_with_resupply(start, target, fleet_id):
+    path_information = namedtuple('path_information', ['distance', 'fuel', 'path'])
+    universe = fo.getUniverse()
+    empire_id = fo.empireID()
+    fleet = universe.getFleet(fleet_id)
+    supplied_systems = set(fo.getEmpire().fleetSupplyableSystemIDs)
+
+    # initialize data structures
+    start_fuel = fleet.maxFuel if start in supplied_systems else fleet.fuel
+    path_cache = {}
+    queue = [(path_information(distance=0, fuel=start_fuel, path=tuple()), start)]
```

Comment:
Will look tomorrow. 

> https://docs.python.org/2/library/heapq.html
To create a heap, use a list initialized to [], or you can transform a populated list into a heap via function heapify().

Reply:
Yes it works.  My point that `heapify()` is expected for to be used together with  `heappop`. If someone diside to change this code it is possible to get errors here.

Reply:
Any PR that tries to initialize any pathfinding code with more than the starting code is immediately to be rejected...

Reply:
Why you want to hold this knowledge?  Why not to put it in code?